Sources/MongoSwift/Operations/ListCollectionsOperation.swift

```diff
@@ -0,0 +1,114 @@
+import mongoc
+
+/// Describes the type of data store returned when executing `listCollections`.
+public enum CollectionType: RawRepresentable, Codable {
+    /// Specifies that the data store returned is a collection.
+    case collection
+    /// Specifies that the data store returned is a view.
+    case view
+    /// For an unknown value. For forwards compatibility, no error will be thrown when an unknown value is provided.
+    case other(String)
+
+    public var rawValue: String {
+        switch self {
+        case .collection:
+            return "collection"
+        case .view:
+            return "view"
+        case .other(let v):
+            return v
+        }
+    }
+
+    public init(rawValue: String) {
+        switch rawValue {
+        case "collection":
+            self = .collection
+        case "view":
+            self = .view
+        default:
+            self = .other(rawValue)
+        }
+    }
+}
+
+/// Specifications of a collection returned when executing `listCollections`.
+public struct CollectionSpecification: Codable {
```

Comment:
This struct is missing a few fields that the server returns back, namely `info` and `idIndex`. See the [official docs](https://docs.mongodb.com/manual/reference/command/listCollections/#listCollections.cursor) for more info on them. On a related note, we can add a `seeAlso` with that link here in the docstring for this struct. You can take a look at examples in [`ChangeStream.swift`](https://github.com/mongodb/mongo-swift-driver/blob/master/Sources/MongoSwift/MongoCollection%2BChangeStreams.swift) for what the formatting of that is.

For `info`, instead of using `Document`, we can define another struct (e.g. `CollectionSpecificationInfo`) that has the required fields and conforms to `Codable`.

Reply:
This struct still needs the `seeAlso` link